### DISCUSSION TITLE
Is it normal to have slow speeds on Synology nas devices?

### DISCUSSION THREAD

**Discussion by aliuq** [immich-app/immich#14133] [Q&A]
After uploading 400 pictures, about 400M in size, more than 30 pictures failed to upload, and there were many error messages on the console. Is this normal?

[docker-compose.yml](https://github.com/immich-app/immich/blob/main/docker/docker-compose.yml)

**Comment by alextran1502** [comment]
Which error are you seeing? And no, it is not normal

**Reply by aliuq** [reply]
![image](https://github.com/user-attachments/assets/d8d844ac-b2f5-473d-bef8-26ade74e6cab)

**Reply by alextran1502** [reply]
Is your database on ssd or hdd? And did you change the job concurrency?

**Reply by aliuq** [reply]
Hi，my NAS uses an HDD hard drive and has not modified the settings.

**Reply by alextran1502** [reply]
HDD has slow write speed and io spec, so the database timeout when the write request is made and cannot complete on time.

I suggest to have your OS and database on the SSD to improve the machine performance and experience in general

**Reply by aliuq** [reply]
Hi，NAS devices generally use HDD hard disk storage + SSD hard disk cache

I simply uploaded 400p. Currently, refreshing the page is very slow, but my actual photos are 100,000+

**Reply by mmomjian** [reply]
Unfortunately as Alex said the use of hard drive for the database is the likely cause here, there are ways to use the SSD for storage as well depending on your specific NAS. The database is only a few GB max. What are the CPU/RAM specs of your NAS?

**Reply by aliuq** [reply]
Due to format issues, it is not possible to simply add an SSD hard drive for expansion. I use Synology’s own photos and there is no such problem.

**Reply by mmomjian** [reply]
Sorry to hear that. It's not really fair to compare synology photos to Immich, since Immich has many more features the requirements are greater.

**Reply by mertalev** [reply]
I kind of doubt it's an HDD issue. I think it could be a CPU issue if it's overloaded and can't keep up. This is a Synology...

Try lowering job concurrencies all to 1. Unless you're using hardware acceleration (which would help tremendously), pause transcoding in particular as it's a huge resource hog and could stall other jobs on a weak CPU.
